src/Status.h

```diff
@@ -30,45 +30,69 @@ namespace Status {
 
 /// \class MessageInfo
 /// Stores cumulative information about received messages: number, size (in
-/// Megabytes) and number of errors
+/// Megabytes) and number of errors. Assuming a 1-to-1 mapping between Streamer
+/// and Topic there will be no concurrent updates of the information, so that
+/// members are not required to be atomic. Nevertheless there is concurrency
+/// between writes (Streamer) and reads (Report). If no syncronisation mechanism
+/// would be present there can be a mixing of updated and non-updated
+/// informations, which the mutex allows to avoid.
 class MessageInfo {
 
 public:
-  using value_type = std::pair<double, double>;
-  /// Implement the = operator for MessageInfo.
-  const MessageInfo &operator=(const MessageInfo &Info);
-
-  /// Implement the += operator for MessageInfo using atomic_add
-  const MessageInfo &operator+=(const MessageInfo &Info);
+  MessageInfo() = default;
+  MessageInfo(const MessageInfo &Other) = default;
```

Comment:
The copy and move constructors you define here are implicitly deleted because the class has a `std::mutex` as a data member. Best to remove or explicitly delete these constructors.